actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -55,7 +55,9 @@ def _back_url # :nodoc:
       def _filtered_referrer # :nodoc:
         if controller.respond_to?(:request)
           referrer = controller.request.env["HTTP_REFERER"]
-          if referrer && URI(referrer).scheme != "javascript"
+          request_method = controller.request.env["REQUEST_METHOD"]
+
+          if referrer && request_method == "GET" && URI(referrer).scheme != "javascript"
```

Comment:
Could this use [Rack::Request::Helpers#get?](https://www.rubydoc.info/gems/rack/Rack%2FRequest%2FHelpers:get%3F)?

```suggestion
          if referrer && URI(referrer).scheme != "javascript" && controller.request.get?
```

Reply:
I think in theory, yes, but unfortunately the many of the corresponding tests rely on stubbing out the controller with a `Struct`.  This, in addition to https://github.com/rails/rails/pull/53722#discussion_r1854954896 feel like a good opportunity for a subsequent commit to clean up these tests, but for now I think I'll keep this commit focused on the simplest implementation.